ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -426,11 +445,17 @@ private[kvutils] class ProcessTransactionSubmission(
     val stateKey = contractIdToStateKey(coid)
     for {
       // Fetch the state of the contract so that activeness and visibility can be checked.
-      contractState <- inputState.get(stateKey).flatMap(_.map(_.getContractState)).orElse {
-        logger.warn(
-          s"Lookup contract failed, contractId=$coid correlationId=${transactionEntry.commandId}")
-        throw Err.MissingInputState(stateKey)
-      }
+      contractState <- inputState.get(stateKey).flatMap(_.map(_.getContractState))
+      // PREVIOUSLY: we aborted the validation if the input state for the contract id wasn't loaded,
+      // because it hinted at a missing input declaration. however, this only worked due to a bug
+      // in lookupKey.
+      //
+      // NOW: There is the possibility that the reinterpretation of the transaction yields a different
+      // result in a LookupByKey than the original transaction. This means that we might not have loaded
+      // the contract state data for the contractId pointed to by that contractKey.
+      // This is not a problem because after the transaction reinterpretation, we compare the original
+      // transaction with the reintrepreted one, and the LookupByKey node will not match.
+      // Additionally, all contract keys are checked to uphold causal monotonicity.
       if contractIsActiveAndVisibleToSubmitter(transactionEntry, contractState)
```

Comment:
Please move the explanation to above the `contractState` definition -- the `if` clause should be right after the variable definition (otherwise the connection is not immediately clear between the filter clause and what it filters).

Reply:
Fixed in 14a595f.